### ISSUE TITLE
AIR builds hang on startup with solid blue screen

### ISSUE THREAD

**Issue by dragontamer8740** [OXOIndustries/Trials-in-Tainted-Space#606]
AIR builds hang on startup with solid blue screen.
Security.allowDomain() and Security.allowInsecureDomain() are not legal in AIR.
I've fixed it by changing (in 'TiTS.as'):
```
Security.allowDomain("*");
Security.allowInsecureDomain("*");
```

to
```
CONFIG::FLASH
{
	Security.allowDomain("*");
	Security.allowInsecureDomain("*");
}
```

I don't know if you need to add some AIR-specific code or not, but as it stands it doesn't work properly without this change made.

Don't know if you intended this to use a new launcher/splash thing on the builds, but I'm just saying what I've noticed

**Comment by Gedan** [comment]
For the AIR builds; no. The loader shit has had to turn into some multi-tier monstrosity to get it to support all the things we wanted from it.

**Comment by dragontamer8740** [comment]
Alright. I am not clear what exactly the purpose of the loader is - but it does look pretty fancy.

Glad to see it got fixed ([revision 70c89b2](https://github.com/OXOIndustries/Trials-in-Tainted-Space/commit/70c89b2da96d9cbae4bee03d728fbd8c213a145d)

**Comment by Gedan** [comment]
We quite often had comments from people clicking the play links and then thinking the game was broken as it would just render as a white screen until the entire thing loaded. It would actually work fine, they just don't expect to sit in front of a white screen for ~30 seconds whilst their terrible internet manages to choke down the SWF.

Than in and of itself wasn't the primary reason for building it though; FurAffinity has a maximum size limit for uploaded SWFs that we ran into a few weeks (months?) ago. I built a separate loader SWF that requests the actual game SWF from fenoxo.com and loads it that way, end-running around the upload size limits on FA. By that point the code was written, I just had to make a few tweaks to also crank out a loader-integrated version too.

I still have some work to do on the site to make this all work perfectly fine though; this has re-exposed an issue we had with Chrome's flash handling that we ran into like 2 years ago now- Chrome won't actually render the loader all the time if you view the SWF directly. If you embed it in a page (even if its just a fit-to-window embed), it will work fine though. Some people seem to experience this WAY more than others, and I've thus far been entirely unable to find rhyme or reason for it. With the same exact Chrome release, with same exact settings, one of my computers will experience it like 95% of the time, and the other maybe 5-10%.

**Comment by Gedan** [comment]
tl;dr it turns out we're doing a lot of things that a lot of the tools around Flash seem to hate. One of the latest round of AIR fixes was disabling optimisation of the Flash bytecode for the AIR SWF itself. Don't ask me what the shit we've done that was causing that to choke, but there you go.

**Comment by dragontamer8740** [comment]
Heh, read the whole thing. Glad I don't deal with chrome/chromium. Though my setup using wine and the windows NPAPI plugin in Linux is far from ideal. Could the problem between your computers be hardware acceleration? Could be a bug with acceleration that the Pepper API plugin is hitting hard.
